### PR TITLE
Docs Update for build preferences, prebuilt binaries

### PR DESCRIPTION
**Pull request by luarss**
- Update build preferences: Docker > Prebuilt > Local
- Prebuilt binaries documentation for website + github
- docker_shell utility
@vvbandeira

### CODE REVIEW

File: docs/user/Build.md [hunk 1]
```diff
@@ -1,5 +1,8 @@
 # Build OpenROAD Flow
 
+We support three major ways of installation. They are: [Docker](./BuildWithDocker.md), [Prebuilt Binaries](./BuildWithPrebuilt.md), and finally [local installation](./BuildLocally.md). You may also choose to follow this script to customise your build process.
```
Comment (vijayank88) [inline docs/user/Build.md]: We support following three major ways of installation. 
- [Docker](./BuildWithDocker.md)
- [Prebuilt Binaries](./BuildWithPrebuilt.md)
- [Local Installation](./BuildLocally.md)

You may also choose to follow this script to customise your build process.

File: docs/user/Build.md [hunk 2]
```diff
@@ -1,5 +1,8 @@
 # Build OpenROAD Flow
 
+We support three major ways of installation. They are: [Docker](./BuildWithDocker.md), [Prebuilt Binaries](./BuildWithPrebuilt.md), and finally [local installation](./BuildLocally.md). You may also choose to follow this script to customise your build process. 
+
+
```
Comment (vijayank88) [inline docs/user/Build.md:5]: Remove extra line

File: docs/user/BuildWithPrebuilt.md [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+Step 3: Run the install command based on platform use package installer.
+        For example Ubuntu 20.04 use:
+```shell
+sudo apt install ./openroad_2.0_amd64-ubuntu20.04.deb
+```
+
+You can install these binaries within docker as well.
+
+> **Thanks** to Precision Innovations (www.precisioninno.com) for providing
```
Comment (vijayank88) [inline docs/user/BuildWithPrebuilt.md]: **Thanks** to [Precision Innovations](https://precisioninno.com/) for providing

**Review by vijayank88 — CHANGES_REQUESTED** [review]

**Comment by luarss** [comment]
@vijayank88 Updated.

**Comment by vvbandeira** [comment]
@luarss please fix the DCO check, see the "Details" link for more details.

File: docs/user/BuildWithPrebuilt.md (hunk 3 above)
Comment (vijayank88) [reply docs/user/BuildWithPrebuilt.md]: this is pending

File: docs/user/BuildWithPrebuilt.md [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+> and supporting this daily release of pre built installers to the community.
+
+
```
Comment (vijayank88) [inline docs/user/BuildWithPrebuilt.md:34]: remove extra line

**Review by vijayank88 — CHANGES_REQUESTED** [review]

**Comment by luarss** [comment]
@vvbandeira Should be fixed now. Will remember to commit with -s option from now on.

File: docs/user/BuildWithPrebuilt.md [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+Step 3: Run the install command based on platform use package installer.
+        For example Ubuntu 20.04 use:
+```shell
+sudo apt install ./openroad_2.0_amd64-ubuntu20.04.deb
+```
+
+You can install these binaries within docker as well.
+
+> **Thanks** to [Precision Innovations](https://www.precisioninno.com) for providing
```
Comment (vijayank88) [inline docs/user/BuildWithPrebuilt.md]: Update to 
```
Thanks to [Precision Innovations](https://precisioninno.com/) for providing
```

**Review by vijayank88 — CHANGES_REQUESTED** [review]

File: README.md [hunk 6]
```diff
@@ -36,6 +32,13 @@ To manage docker as non-root user and verify that you can run
 Document for detailed steps on docker based installation found
 [here](./docs/user/BuildWithDocker.md).
 
+### Prebuilt Binaries
+Document for installing prebuilt binaries courtesy of Precision Innovations found [here](./docs/user/BuildWithPrebuilt.md).
```
Comment (vvbandeira) [inline README.md]:
```suggestion
### Prebuilt Binaries

There are third-party vendors that distribute prebuilt binary versions of the applications required by ORFS. OpenROAD app, Yosys and Klayout. Note that the version you will install from a third party is not guaranteed to work with any version of ORFS. See instructions [here](./docs/user/BuildWithPrebuilt.md).
```

Comment (vvbandeira) [reply README.md]: @dralabeing please advise on the wording.

File: docs/user/BuildWithPrebuilt.md [hunk 7]
```diff
@@ -0,0 +1,28 @@
+# Build from sources using Prebuilt Binaries
```
Comment (vvbandeira) [inline docs/user/BuildWithPrebuilt.md]: This looks to be a copy of the text from the OR docs. We should always avoid having the same instructions in two different places as this makes it more difficult to keep up-to-date. Please link to OR-docs instead. Also, here we will need to add instructions on how to install Yosys and Klayout using `apt`/`.deb` files. Please also double-check if the user needs to set any special PATH variable to run `make` commands after installing the prebuilt binaries.

File: docs/user/BuildLocally.md [hunk 8]
```diff
@@ -46,40 +46,9 @@ make gui_final
...
-
-> **Thanks** to Precision Innovations (www.precisioninno.com) for providing
-> and supporting this daily release of pre built installers to the community.
-
 ## Compiling and debugging in Visual Studio Code
 
-Set up environment variables to point to tools that CMake from within
-Visual Studio Code will need, then start Visual Studio Code as usual
-and hit F7, assuming you have CMake plugins installed.
+Set up environment variables using this script, then start Visual Studio Code. Please ensure [CMake plugins](https://code.visualstudio.com/docs/cpp/cmake-linux) are installed.
```
Comment (vvbandeira) [inline docs/user/BuildLocally.md]: > this script

The statement is vague; please be explicit on the script that should be used.
Please also change to:

> Please ensure that the [CMake plugin](https://code.visualstudio.com/docs/cpp/cmake-linux) is installed.

File: docs/user/Build.md [hunk 9]
```diff
@@ -1,5 +1,13 @@
 # Build OpenROAD Flow
 
+We support three major ways of installation.
```
Comment (vvbandeira) [inline docs/user/Build.md]:
```suggestion
We support three major ways of installation:
```

File: docs/user/BuildWithDocker.md [hunk 10]
```diff
@@ -72,6 +72,41 @@ docker run --rm -it \
            openroad/flow-$OS_NAME-builder
 ```
 
+
+### Docker shell utility
```
Comment (vvbandeira) [inline docs/user/BuildWithDocker.md]: This is also very similar to the README file in the `flow/util` folder; please refer to the comment below on not duplicating text.

**Review by vvbandeira — CHANGES_REQUESTED** [review]

File: docs/user/BuildWithPrebuilt.md (hunk 7 above)
Comment (luarss) [reply docs/user/BuildWithPrebuilt.md]: @vvbandeira I think the concern was that we wanted to separate the prebuilt binaries install instructions from built local instructions. Should I modify the OR docs as well?

File: docs/user/BuildWithDocker.md (hunk 10 above)
Comment (luarss) [reply docs/user/BuildWithDocker.md]: @vvbandeira Noted. Where in the toc do you think we should slot the flow/util doc page?

File: docs/user/BuildWithPrebuilt.md (hunk 7 above)
Comment (vvbandeira) [reply docs/user/BuildWithPrebuilt.md]: (I think we should remove the `Build from sources` as it does not make much sense with prebuilt bins)

Adding the new file is correct, but we do not want to duplicate the instructions from OR docs here. This new file (i.e., `docs/user/BuildWithPrebuilt.md`) should have sections similar to the following:

```
# Using Prebuilt Binaries

## Install Yosys and Klayout

Either: apt install -y yosys klayout
Or the necessary steps to download and install, will need to double-check their docs.

## Install OpenROAD

[link to OR docs]

## Aditional setup

> might be necessary, not sure
```

Comment (vvbandeira) [reply docs/user/BuildWithPrebuilt.md]: Ping me on slack if a quick meeting would be better to understand.

File: docs/user/BuildWithDocker.md (hunk 10 above)
Comment (vvbandeira) [reply docs/user/BuildWithDocker.md]: I think a sub-entry bellow the build with docker

File: README.md (hunk 6 above)
Comment (dralabeing) [reply README.md]: We are guaranteeing pre-built binaries from Precision but not from other third party vendors. So we should not lump this disclaimer into one category--the user may interpret Precision Innovations to also be third party. Hence need to clarify this point.

The link for pre-built binaries should be referred to as follows:
"To download pre-built binaries for ORFS, refer to the documentation link [here]". Also, retain the note thanking Precision Innovations -- only one instance as discussed.

The disclaimer can then be worded as follows: There are several third-party vendors that distribute pre-built versions of the applications required by ORFS including the OpenROAD application, Yosys and Klayout. Note that the version you will install from a third-party other than Precision Innovations is not guaranteed to work with any version of ORFS. See instructions [here] (./docs/user/BuildWithPrebuilt.md).

File: docs/user/BuildWithPrebuilt.md (hunk 7 above)
Comment (dralabeing) [reply docs/user/BuildWithPrebuilt.md]: Agree with Vitor-Remove Build from sources for pre-built. also change the spelling to Pre-built not prebuilt

File: docs/user/BuildWithPrebuilt.md [hunk 11]
```diff
@@ -0,0 +1,48 @@
+# Using Pre-built Binaries
+
+## Install Klayout and Yosys
+Please ensure the Klayout version (denoted with `klayoutVersion` variable) is consistent with the one used in [DependencyInstaller script](https://github.com/The-OpenROAD-Project/OpenROAD-flow-scripts/blob/master/etc/DependencyInstaller.sh). 
+
+Instructions for building:
+- [Klayout>=0.28.8](https://www.klayout.de/build.html)
+- [Yosys>=0.21](https://github.com/YosysHQ/oss-cad-suite-build#installation)
```
Comment (vvbandeira) [inline docs/user/BuildWithPrebuilt.md]:
```suggestion
Instructions for installing:
- [Klayout>=0.28.8](https://www.klayout.de/build.html)
- [Yosys>=0.21](https://github.com/YosysHQ/oss-cad-suite-build#installation)
```

**Comment by luarss** [comment]
@vvbandeira Fixed more missing links. I feel we can start a new PR for future docs update, this PR is getting quite large already. 

For OR docs update, these are the links that might be broken 
image

**Comment by vvbandeira** [comment]
> @vvbandeira Fixed more missing links. I feel we can start a new PR for future docs update, this PR is getting quite large already.


I agree.


> For OR docs update, these are the links that might be broken image

Please create a companion PR in the OR repo to fix these links.

**Comment by vvbandeira** [comment]
@dralabeing, these changes look good to me, we can always iterate and improve later. Please approve if you agree.

**Comment by dralabeing** [comment]
Looks good - please go ahead and merge

**Review by dralabeing — APPROVED** [review]

File: docs/contrib/PlatformBringUp.md [hunk 12]
```diff
@@ -40,7 +40,7 @@ They include :
 Adding a new platform additionally requires the following:
 
 *   A validated installation of the OpenROAD flow scripts is available. See
-    instructions [here](../user/GettingStarted.md).
+    instructions [here](../user/FlowVariables.md).
```
Comment (vijayank88) [inline docs/contrib/PlatformBringUp.md]: @luarss 
Why installation instructions pointing to flow variables list?

**Review by vijayank88 — CHANGES_REQUESTED** [review]

File: docs/contrib/GettingInvolved.md [hunk 13]
```diff
@@ -75,11 +75,11 @@ project like OpenROAD with many application subcomponents, the software
...
 dependencies which are not necessary are less likely to be integrated.
 
-If you want to add Tcl code to define a new tool command, look at pdngen
+If you want to add Tcl code to define a new tool command, look at [pdngen](https://github.com/The-OpenROAD-Project/OpenROAD/tree/master/src/pdn)
 as an example of how to do so. Take a look at the
-[cmake file](https://github.com/The-OpenROAD-Project/OpenROAD/blob/master/src/CMakeLists.txt)
+[CMakeLists file](https://github.com/The-OpenROAD-Project/OpenROAD/blob/master/src/CMakeLists.txt)
 which automatically sources the Tcl code and the
-[Tcl file](https://github.com/The-OpenROAD-Project/OpenROAD/blob/master/src/pdn/src/PdnGen.tcl)
+[Tcl file](https://github.com/The-OpenROAD-Project/OpenROAD/blob/master/src/drt/src/TritonRoute.tcl)
```
Comment (vijayank88) [inline docs/contrib/GettingInvolved.md]: Why referring `pdngen` tool and pointing `TritonRoute.tcl` here?
Keep same tool reference

**Review by vijayank88 — CHANGES_REQUESTED** [review]

**Comment by luarss** [comment]
@vijayank88 Good catch. Do let me know if there are any more errors.

**Comment by luarss** [comment]
@vvbandeira No changes need to be made with the OR docs. I think the user/BuildLocally.html path still exists in the final build.

edit: I looked closer and I think the OR build instructions should not to ORFS. We actually want the OR build instructions to link to its own OR build. I will create one PR on OR docs.

File: docs/user/BuildWithPrebuilt.md [hunk 14]
```diff
@@ -0,0 +1,48 @@
+# Using Pre-built Binaries
+
+## Install Klayout and Yosys
+Please ensure the Klayout version (denoted with `klayoutVersion` variable) is consistent with the one used in [DependencyInstaller script](https://github.com/The-OpenROAD-Project/OpenROAD-flow-scripts/blob/master/etc/DependencyInstaller.sh). 
+
+Instructions for building:
```
Comment (vvbandeira) [inline docs/user/BuildWithPrebuilt.md]:
```suggestion
Instructions for installing:
```

**Pull request merged**